ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs

```diff
@@ -178,10 +178,10 @@ protocolInfoShelley genesis protVer mbCredentials =
       }
 
     initConsensusState :: State.TPraosState c
-    initConsensusState = State.empty $
+    initConsensusState = State.empty Origin $
       SL.PrtclState
          (SL.chainOCertIssue     initShelleyState)
-         Origin
+         SL.NeutralNonce
```

Comment:
Let's mark this as having to be updated (hash of genesis or, if we must, final Byron hash)